sidecar/resources.py

```diff
@@ -151,8 +151,12 @@ def _watch_resource_iterator(label, labelValue, targetFolder, url, method, paylo
         # Check if it's an empty ConfigMap or Secret
         dataMap = event["object"].data
         if dataMap is None:
-            print(f"{timestamp()} {resource} does not have data.")
-            continue
+            
+            dataMap = event["object"].binary_data
```

Comment:
This code now prefers `data` over `binary_data` and in contrast to `listResources` does not operate on both at the same time. I think we should evaluate both. Probably by looping over line 153 to 186 once for `data` and once for `binary_data` or merging the both into a single dict. I think keys present in both are conflicting anyway because are resolved to the same file name so merging might be easier.